resume-buddy/src/main/java/com/google/sps/servlets/BlobstoreServeServlet.java

```diff
@@ -25,19 +22,24 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Gets the logged in user's email, finds the PDF blob and serves it
+    // Gets the logged in user's email, finds the PDF blob from the Match datastore and serves it
     UserService userService = UserServiceFactory.getUserService();
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     FetchOptions fetchOptions = FetchOptions.Builder.withLimit(Integer.MAX_VALUE);
     String email = userService.getCurrentUser().getEmail();
-    Query query = new Query("Reviewee");
-    Filter emailFilter = new FilterPredicate("email", FilterOperator.EQUAL, email);
-    query.setFilter(emailFilter);
+
+    Query query = new Query("Match");
     PreparedQuery results = datastore.prepare(query);
     List<Entity> entityList = results.asList(fetchOptions);
-    // TODO: (sesexton) Create a function that will grab the most recent instead of the first entity
-    // returned
-    String userBlobKeyString = entityList.get(0).getProperty("resumeBlobKey").toString();
+    String userBlobKeyString = "";
+
+    for (Entity pair : entityList) {
+      if ((pair.getProperty("reviewee").toString().equals(email))
+          || (pair.getProperty("reviewer").toString().equals(email))) {
+        userBlobKeyString = pair.getProperty("resumeBlobKey").toString();
+      }
+    }
```

Comment:
I think it would probably be better to do two filtered queries instead of iterating over every single match in the datastore. The first query could use an `EQUAL` filter between `"reviewee"` and `email`. If that turns up empty, you can then run a second query with a filter between `"reviewer"` and `email`.

```
Query query = new Query("Match");
Filter filter = new FilterPredicate("reviewee", FilterOperator.EQUAL, email);
query.setFilter(filter);
PreparedQuery results = datastore.prepare(query);
if (results.countEntities(FetchOptions.Builder.withDefaults()) == 0) {
  // Make a new Query and new Filter here
  results = datastore.prepare(query);
}

// iterate over results...
```

Reply:
Yes, users should only be matched once. At least if/when we are able to implement and option that allows the reviewer to have multiple reviewee's.